dwarf/line/state_machine.go

```diff
@@ -151,9 +161,10 @@ func execSpecialOpcode(sm *StateMachine, instr byte) {
 
 	sm.lastDelta = int(sm.dbl.Prologue.LineBase + int8(decoded%sm.dbl.Prologue.LineRange))
 	sm.line += sm.lastDelta
-	sm.address += uint64(decoded / sm.dbl.Prologue.LineRange)
+	sm.address += uint64(decoded/sm.dbl.Prologue.LineRange) * uint64(sm.dbl.Prologue.MinInstrLength)
```

Comment:
This change is just a formality MinInstrLen, at least on amd64, is always 1. AFAIK delve always interpreted special opcodes correctly:

```
objdump --dwarf ./testvariables2 | grep -A11 'debug_line' 
Raw dump of debug contents of section .debug_line:

  Offset:                      0x0
  Length:                      76002
  DWARF Version:               2
  Prologue Length:             7461
  Minimum Instruction Length:  1
  Initial value of 'is_stmt':  1
  Line Base:                   -4
  Line Range:                  10
  Opcode Base:                 10

```

plugging these values into that code we get https://play.golang.org/p/PhTBbfJ8GX :

```249 = addr 23 line 5
255 = addr 24 line 1
```

I don't know why link would need a special branch to handle 249 and 255. What we interpreted wrong was the standard opcode DW_LNS_const_add_pc, which wasn't emitted (or was only emitted rarely) by previous versions of link.